src/data-handler/index.js

```diff
@@ -43,13 +31,7 @@ export default function createDataHandler({ layoutService, model, extremumModel,
 
       requestInProgress
         .then(() => {
-          requestInProgress = null;
           meta.isBinnedData = !layoutService.getDataPages().length;
-          if (nextInLine) {
-            const tempRef = nextInLine;
-            nextInLine = null;
-            tempRef.resolve(dataHandler.fetch());
-          }
         })
         .catch(() => {});
```

Comment:
I know this `catch` is not new, but If anything goes wrong and throws an error somewhere in the promise chain, even developer errors like misspelling or using something wrong, it will get swallowed up and hidden by this catch.

The promises that are rejected by choice, for example `nextInLine.reject();`, should reject with a rejection token:

```js
import { REJECTION_TOKEN } from '../somewhere/constants';
...
nextInLine.reject(REJECTION_TOKEN);
...

```
Where `REJECTION_TOKEN` can be something simple as `export const REJECTION_TOKEN = 'REJECTED';`.

Then you can swallow those ones, and let the other bubble up:

```js
.catch((e) => {
  if (e !== REJECTION_TOKEN) {
    throw e;
  }
});
```
@veinfors @jjxie @quanho

Reply:
good advice!